frontend/widgets/AudioMixer.cpp

```diff
@@ -0,0 +1,1056 @@
+	if (main) {
+		connect(main, &OBSBasic::userSettingChanged, this,
+			[this](const std::string &category, const std::string &name) {
+				if (category == "BasicWindow" && name == "VerticalVolumeControl") {
+					toggleLayout();
+				} else if (category == "BasicWindow" && name == "MixerShowInactive") {
+					updateShowInactive();
+				} else if (category == "BasicWindow" && name == "MixerKeepInactiveLast") {
+					updateKeepInactiveLast();
+				} else if (category == "BasicWindow" && name == "MixerShowHidden") {
+					updateShowHidden();
+				} else if (category == "BasicWindow" && name == "MixerKeepHiddenLast") {
+					updateKeepHiddenLast();
+				} else if (category == "BasicWindow" && name == "ShowListboxToolbars") {
+					updateShowToolbar();
+				} else if (category == "Accessibility" && name == "SettingsChanged") {
+					refreshVolumeColors();
+				}
+			});
+
+		connect(main, &OBSBasic::mixerStatusChanged, this, &AudioMixer::queueLayoutUpdate);
+
+		connect(advAudio, &QAction::triggered, main, &OBSBasic::on_actionAdvAudioProperties_triggered,
+			Qt::DirectConnection);
+		connect(toggleHiddenButton, &QPushButton::clicked, this, &AudioMixer::toggleShowHidden);
+		connect(layoutButton, &QAction::triggered, main, &OBSBasic::toggleMixerLayout);
+	}
+
+	updateShowToolbar();
+	updatePreviewSources();
+	updateGlobalSources();
+
+	reloadVolumeControls();
+}
+
+AudioMixer::~AudioMixer()
+{
+	signalHandlers.clear();
+
+	previewSources.clear();
+	globalSources.clear();
+
+	clearVolumeControls();
+
+	obs_frontend_remove_event_callback(AudioMixer::onFrontendEvent, this);
+}
+
+void AudioMixer::toggleLayout()
+{
+	bool vertical = config_get_bool(App()->GetUserConfig(), "BasicWindow", "VerticalVolumeControl");
+	setMixerLayoutVertical(vertical);
+
+	queueLayoutUpdate();
+}
+
+void AudioMixer::toggleShowInactive(bool checked)
+{
+	config_set_bool(App()->GetUserConfig(), "BasicWindow", "MixerShowInactive", checked);
+	OBSBasic *main = OBSBasic::Get();
+	if (main) {
+		emit main->userSettingChanged("BasicWindow", "MixerShowInactive");
+	}
+}
+
+void AudioMixer::toggleKeepInactiveLast(bool checked)
+{
+	config_set_bool(App()->GetUserConfig(), "BasicWindow", "MixerKeepInactiveLast", checked);
+	OBSBasic *main = OBSBasic::Get();
+	if (main) {
+		emit main->userSettingChanged("BasicWindow", "MixerKeepInactiveLast");
+	}
+}
+
+void AudioMixer::toggleShowHidden(bool checked)
+{
+	config_set_bool(App()->GetUserConfig(), "BasicWindow", "MixerShowHidden", checked);
+	OBSBasic *main = OBSBasic::Get();
+	if (main) {
+		emit main->userSettingChanged("BasicWindow", "MixerShowHidden");
+	}
+}
+
+void AudioMixer::toggleKeepHiddenLast(bool checked)
+{
+	config_set_bool(App()->GetUserConfig(), "BasicWindow", "MixerKeepHiddenLast", checked);
+	OBSBasic *main = OBSBasic::Get();
+	if (main) {
+		emit main->userSettingChanged("BasicWindow", "MixerKeepHiddenLast");
+	}
+}
+
+VolumeControl *AudioMixer::createVolumeControl(obs_source_t *source)
```

Comment:
This function (and many functions it calls) are all part of the `AudioMixer` public API, but they don't seem to interact with the object instance at all.

A `create<Something>` function like this is typical for a factory pattern and is usually either a free-standing function in a named namespace or a `static` function of the class.